engine/apps/public_api/serializers/integrations.py

```diff
@@ -166,30 +169,155 @@ def validate_templates(self, templates):
                 raise BadRequest(detail=f"Invalid {common_template} template data")
         return templates
 
-    def _correct_validated_data_for_messaging_backends(self, validated_data):
-        templates = validated_data.get("templates", {})
+    def _correct_validated_data(self, validated_data):
+        """
+        Process input templates data.
+        1. Reshapes it.
+          1.1 We are receiving templates in dict format
+            {
+              resolve_signal: "resolve me!"
+              slack: {
+                title: "title",
+                message: "message",
+                image_url: "image_url",
+              },
+              ...
+            }
+          but store them in separate fields: slack_title_template, slack_message_template.
+          See _correct_validated_data_for_legacy_template method
+
+          1.2 We are storing templates from messaging backends in separate messaging_backends_templates field.
+          So, we need to shape input data related to messaging_backends_templates also.
+        2. Handle None templates.
+         Public API set template to default value in two cases: (This behaviour is required by terraform plugin).
+         2.1 None for the whole template:
+         {
+           slack: None,
+           ...
+         }
+         In that case all slack templates will be set to default.
+
+         2.2 One particular field is None:
+         {
+           slack: {
+             title: "My custom title:
+             message: None,
+           },
+           ...
+         }
+         In that case slack message template will be set to default.
+
+        TODO: System described above is too complicated, should be simplified.
+        It can be simplified via unification all chatops integrations and messaging_backends
+        and/or by introducing unified templates
+        """
 
-        messaging_backends_templates = self.instance.messaging_backends_templates if self.instance else None
+        validated_data = self._correct_validated_data_for_messaging_backends_templates(validated_data)
 
-        for backend_id, backend in get_messaging_backends():
-            backend_templates = {}
-            if messaging_backends_templates is not None:
-                backend_templates = messaging_backends_templates.get(backend_id, {})
+        # backend_slug_to_backend = {}
+        # for backend_id, backend in get_messaging_backends():
+        #     backend_slug_to_backend[backend.backend_slug] = backend
 
-            for field in backend.template_fields:
+        templates_data_from_request = validated_data.get("templates", {})
+        for template_backend_name, templates_data_from_request in templates_data_from_request.items():
```

Comment:
Moving this loop to `_correct_validated_data_for_legacy_templates` would make this easier to understand:

```
def _correct_validated_data(self, validated_data):
    validated_data = self._correct_validated_data_for_messaging_backends_templates(validated_data)
    validated_data = self._correct_validated_data_for_legacy_templates(validated_data)
```